src/appendix/glossary.md

```diff
@@ -1,96 +1,491 @@
-<span id="ut">uninhabited type</span>          |  A type which has _no_ values. This is not the same as a ZST, which has exactly 1 value. An example of an uninhabited type is `enum Foo {}`, which has no variants, and so, can never be created. The compiler can treat code that deals with uninhabited types as dead code, since there is no such value to be manipulated. `!` (the never type) is an uninhabited type. Uninhabited types are also called "empty types".
-<span id="upvar">upvar</span>                  |  A variable captured by a closure from outside the closure.
-<span id="variance">variance</span>            |  Determines how changes to a generic type/lifetime parameter affect subtyping; for example, if `T` is a subtype of `U`, then `Vec<T>` is a subtype `Vec<U>` because `Vec` is *covariant* in its generic parameter. See [the background chapter](./background.md#variance) for a more general explanation. See the [variance chapter](../variance.md) for an explanation of how type checking handles variance.
-<span id="variant-idx">variant index</span>    |  In an enum, identifies a variant by assigning them indices starting at 0.  This is purely internal and not to be confused with the ["discrimiant"](#discriminant) which can be overwritten by the user (e.g. `enum Bool { True = 42, False = 0 }`).
-<span id="wide-ptr">wide pointer</span>        |  A pointer with additional metadata. See "fat pointer" for more.
-<span id="zst">ZST</span>                      |  Zero-Sized Type. A type whose values have size 0 bytes. Since `2^0 = 1`, such types can have exactly one value. For example, `()` (unit) is a ZST. `struct Foo;` is also a ZST. The compiler can do some nice optimizations around ZSTs.
-
-[LLVM]: https://llvm.org/
+#### arena/arena allocation
+
+An _arena_ is a large memory buffer from which other memory allocations are
+made. This style of allocation is called _arena allocation_. See [this
+chapter](../memory.md) for more info.
+
+#### AST
+
+The abstract syntax tree produced by the `rustc_ast` crate; reflects user syntax
+very closely.
+
+#### binder
+
+A "binder" is a place where a variable or type is declared; for example, the
+`<T>` is a binder for the generic type parameter `T` in `fn foo<T>(..)`, and
+\|`a`\|`...` is a binder for the parameter `a`. See [the background chapter for
+more](./background.md#free-vs-bound).
+
+#### BodyId
+
+An identifier that refers to a specific body (definition of a function or
+constant) in the crate. See [the HIR chapter for
+more](../hir.md#identifiers-in-the-hir).
+
+#### bound variable
+
+A "bound variable" is one that is declared within an expression/term. For
+example, the variable `a` is bound within the closure expression \|`a`\|`a * 2`.
+See [the background chapter for more](./background.md#free-vs-bound)
+
+#### codegen
+
+The code to translate MIR into LLVM IR.
```

Comment:
This could maybe be added, but I'm not even sure it's correct. No need to change this for now
```suggestion
The code to translate MIR into LLVM IR. Short for "code generation".
```